### ISSUE TITLE
[Bug] AbsoluteLayout incorrectly checks bounds.Width instead of Height

### ISSUE THREAD

**Issue by KPixel** [xamarin/Xamarin.Forms#10356]
### Description

This is a minor bug that slipped through for years:

[The line 254 of AbsoluteLayout](https://github.com/xamarin/Xamarin.Forms/blob/4a2a14e33f0ac9b7a691690b5f9b12ecf6bafa97/Xamarin.Forms.Core/AbsoluteLayout.cs#L254) is currently:

```CSharp
if (!heightIsProportional && bounds.Width == AutoSize)
```
But, it should be:

```CSharp
if (!heightIsProportional && bounds.Height == AutoSize)
```

### Basic Information

- Version with issue: All (in this repo, [see the very first commit](https://github.com/xamarin/Xamarin.Forms/blob/17fdde66d94155fc62a034fa6658995bef6fd6e5/Xamarin.Forms.Core/AbsoluteLayout.cs#L241))

**Comment by samhouts** [comment]
@KPixel Do you have a sample project that demonstrates that this is a bug? Thanks!

**Comment by KPixel** [comment]
No, but it is possible to create such a sample with a view that has an "auto size" width and a fixed height. In that case, the height will be ignored.

Reading the following section of code, it is clear that the lines 263-264 will not be triggered (as they should). And it is obvious that the condition `bounds.Width == AutoSize` at line 254 is always true (since it was already checked at line 252).

https://github.com/xamarin/Xamarin.Forms/blob/4a2a14e33f0ac9b7a691690b5f9b12ecf6bafa97/Xamarin.Forms.Core/AbsoluteLayout.cs#L252-L273

**Comment by jsuarezruiz** [comment]
Attached a reproduction sample: [Issue10356.zip](https://github.com/xamarin/Xamarin.Forms/files/4509038/Issue10356.zip)

**Comment by KPixel** [comment]
Thanks for the sample.

> Seems that heightIsProportional in line 254 must be widthtIsProportional.

Wouldn't that make line 254 to be identical to line 252?
I think the correct fix is the one mentioned in my initial post: `bounds.Height == AutoSize`

**Comment by KPixel** [comment]
Can you please test this sample on UWP as well?
It doesn't work there.
